ui/server.jsx

```diff
@@ -0,0 +1,40 @@
+/* eslint no-console: 'off' */
+import path from 'path';
+
+import cfenv from 'cfenv';
+import express from 'express';
+import morgan from 'morgan';
+import React from 'react';
+import { renderToString } from 'react-dom/server';
+import { match, RouterContext } from 'react-router';
+
+import routes from './routes';
+
+const app = express();
+const env = cfenv.getAppEnv();
+
+app.use(morgan('combined'));
+app.use('/static', express.static(path.join('ui-dist', 'static')));
+
+app.get('*', (req, res) => {
```

Comment:
TODO: are there any libraries or patterns that make this a bit neater? I know that Node tends to get messy this way. Definitely not something to hold up merge for, especially since this is a first step. It feels wrong to handle the response and define the base HTML in the same place, but I'm also new to JSX. This may very well be the pattern!